protocols/directfund/directfund.go

```diff
@@ -56,6 +56,14 @@ type GetTwoPartyConsensusLedgerFunction func(counterparty types.Address) (ledger
 
 // NewObjective creates a new direct funding objective from a given request.
 func NewObjective(request ObjectiveRequest, preApprove bool, myAddress types.Address, chainId *big.Int, getChannels GetChannelsByParticipantFunction, getTwoPartyConsensusLedger GetTwoPartyConsensusLedgerFunction) (Objective, error) {
+	channelExists, err := channelsExistWithCounterparty(request.CounterParty, getChannels, getTwoPartyConsensusLedger)
+	if err != nil {
+		return Objective{}, fmt.Errorf("counterparty check failed: %w", err)
+	}
+	if channelExists {
+		return Objective{}, fmt.Errorf("a channel already exists with counterparty %s", request.CounterParty)
+	}
+
```

Comment:
This check is essentially a precondition for creating a directfund. The change here is that it was moved to top of function instead of after some work is done.

Reply:
Side note: there are possible futures where hub-hub users may want to have multiple ledger channels between them for capacity reasons (eg, max number of running guarantees).